test/integration/functional_test.go

```diff
@@ -96,6 +99,7 @@ func TestFunctional(t *testing.T) {
 			{"TunnelCmd", validateTunnelCmd},
 			{"SSHCmd", validateSSHCmd},
 			{"MySQL", validateMySQL},
+			{"FileSync", validateFileSync},
```

Comment:
@nanikjava the problem with this test is it starts a new minikube just for testing file sync. which will add time to the integraiton tests.


what I recommend is moving your test to 

this part of the test
```
	// Serial tests
	t.Run("serial", func(t *testing.T) {
		tests := []struct {
			name      string
			validator validateFunc
		}{
			{"StartWithProxy", validateStartWithProxy}, // Set everything else up for success
			{"KubeContext", validateKubeContext},       // Racy: must come immediately after "minikube start"
			{"KubectlGetPods", validateKubectlGetPods}, // Make sure apiserver is up
			{"CacheCmd", validateCacheCmd},             // Caches images needed for subsequent tests because of proxy
		}
```

in the abve code we have 4 serial tests, but only one of them starts minikube ! (start proxy)

what I recommend is, you have s function that is fileSyncTestSetup
that runs _before_ . validateStartWithProxy

and then you have a test _after "validateStartWithProxy" test in that checks if the file is synced.
(then you can delete the "start" in your test

Reply:
In the normal user scenario, the file sync process in minikube gets triggered only by doing 'minikube start' so in the test case the function 'validateFileSync' is doing exactly that. The 'serial' test starts minikube which is fine but in order for the file sync process to kicks in the test case need to simulate 'minikube start'. That is what I'm trying to simulate in the test case

Correct me if I'm wrong but from your comment you are suggesting to copy the files locally _before_ the 'minikube start' test case starts ?.

Reply:
Agree with @medyagh here, even if it's a bit weird. `TestFunctional/parallel` is designed for tests that can be run in parallel and share a single minikube cluster for both performance and resource consumption reasons. We should make the comments clearer here.

I second medya's suggestion. Move the file creation to `validateStartWithProxy`, where it will then then be later available in `validateFileSync`.

Reply:
After re-reading the comments I think I understand what need to be done. Will fix it up. Thanks @medyagh and @tstromberg for the comments

Reply:
Fix the PR